tutorials/llm/llama-3/nemo2-sft-peft/nemo2-sft.ipynb

```diff
@@ -0,0 +1,632 @@
+    "\n",
+    "For more information, refer to [databricks-dolly-15k | Hugging Face](https://huggingface.co/datasets/databricks/databricks-dolly-15k)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "# Step 1: Import HuggingFace checkpoint\n",
+    "We use the `llm.import_ckpt` API to download the specified model using the \"hf://<huggingface_model_id>\" URL format. It will then convert the model into NeMo 2.0 format. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import nemo_run as run\n",
+    "from nemo import lightning as nl\n",
+    "from nemo.collections import llm\n",
+    "from megatron.core.optimizer import OptimizerConfig\n",
+    "import torch\n",
+    "import pytorch_lightning as pl\n",
+    "from pathlib import Path\n",
+    "from nemo.collections.llm.recipes.precision.mixed_precision import bf16_mixed\n",
+    "\n",
+    "\n",
+    "# llm.import_ckpt is the nemo2 API for converting Hugging Face checkpoint to NeMo format\n",
+    "# example python usage:\n",
+    "# llm.import_ckpt(model=llm.llama3_8b.model(), source=\"hf://meta-llama/Meta-Llama-3-8B\")\n",
+    "#\n",
+    "# We use run.Partial to configure this function\n",
+    "def configure_checkpoint_conversion():\n",
+    "    return run.Partial(\n",
+    "        llm.import_ckpt,\n",
+    "        model=llm.llama3_8b.model(),\n",
+    "        source=\"hf://meta-llama/Meta-Llama-3-8B\",\n",
+    "        overwrite=False,\n",
+    "    )\n",
+    "\n",
+    "# configure your function\n",
+    "import_ckpt = configure_checkpoint_conversion()\n",
+    "# define your executor\n",
+    "local_executor = run.LocalExecutor()\n",
+    "\n",
+    "# run your experiment\n",
+    "run.run(import_ckpt, executor=local_executor)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Prepare data and customize DataModule\n",
+    "\n",
+    "We will be using Databricks-dolly-15k for this notebook. NeMo 2.0 already provides a `DollyDataModule`. Example usage:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def dolly() -> run.Config[pl.LightningDataModule]:\n",
+    "    return run.Config(llm.DollyDataModule, seq_length=2048, micro_batch_size=1, global_batch_size=8, num_workers=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To use your own data, you will need to create a custom `DataModule`. This involves extending the base class `FineTuningDataModule`, so that you have access to existing data handling logic such as packed sequence. Here we walk you through the process step by step using the already existing [`DollyDataModule`](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/llm/gpt/data/dolly.py) as an example. \n",
```

Comment:
fix punctuation

 "To use your own data, you will need to create a custom `DataModule`. This involves extending the base class `FineTuningDataModule`, so that you have access to existing data handling logic, such as packed sequences. Here we walk you through the process step by step, using the already existing [`DollyDataModule`](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/llm/gpt/data/dolly.py) as an example. \n",